packages/catalog-search/src/SearchSuggestionItem.jsx

```diff
@@ -3,43 +3,60 @@ import { Link } from 'react-router-dom';
 import PropTypes from 'prop-types';
 
 const SearchSuggestionItem = ({
-  title, url, highlightedTitle, authoringOrganization, programType,
-}) => (
-  <Link to={url} key={title} className="suggestion-item">
-    <div>
-      {/* eslint-disable-next-line react/no-danger */}
-      <div dangerouslySetInnerHTML={{ __html: highlightedTitle }} />
-      {
+  title, url, highlightedTitle, authoringOrganization, programType, handleSubmit, disableSuggestionRedirect, hit,
+}) => {
+  // If the disable redirect bool is provided, prevent the redirect from happening and instead call the provided submit
+  // handler
+  const handleLinkDisable = (e) => {
+    if (disableSuggestionRedirect) {
+      e.preventDefault();
+      handleSubmit(hit);
+    }
+  };
+
+  return (
+    <Link to={url} key={title} className="suggestion-item" onClick={handleLinkDisable}>
+      <div>
+        { /* eslint-disable-next-line react/no-danger */ }
+        <div dangerouslySetInnerHTML={{ __html: highlightedTitle }} />
+        {
           authoringOrganization && (
             <div className="badge badge-light ml-3 font-weight-light authoring-org-badge">
-                {authoringOrganization}
+              {authoringOrganization}
             </div>
           )
+        }
+      </div>
+      {
+        programType && (
+          <p className="font-weight-light text-gray-400 program-type">
+            {programType}
+          </p>
+        )
       }
-    </div>
-    {
-      programType && (
-        <p className="font-weight-light text-gray-400 program-type">
-          {programType}
-        </p>
-      )
-    }
-
-  </Link>
-);
+    </Link>
+  );
+};
 
 SearchSuggestionItem.propTypes = {
   title: PropTypes.string.isRequired,
   url: PropTypes.string.isRequired,
   highlightedTitle: PropTypes.string.isRequired,
   authoringOrganization: PropTypes.string,
   programType: PropTypes.string,
+  handleSubmit: PropTypes.func.isRequired,
+  disableSuggestionRedirect: PropTypes.bool.isRequired,
+  hit: PropTypes.shape({
```

Comment:
yeah same thought here: given hit already has a bunch of the needed fields, it seems now the props like title and authoringOrganization feel superflous (also more props means it gets harder to test the component over time)